bgpd/bgp_evpn.c

```diff
@@ -2164,26 +2164,24 @@ static int delete_global_type2_routes(struct bgp *bgp, struct bgpevpn *vpn)
 
 	rddest = bgp_node_lookup(bgp->rib[afi][safi],
 				 (struct prefix *)&vpn->prd);
-	if (rddest && bgp_dest_has_bgp_path_info_data(rddest)) {
-		table = bgp_dest_get_bgp_table_info(rddest);
-		for (dest = bgp_table_top(table); dest;
-		     dest = bgp_route_next(dest)) {
-			const struct prefix_evpn *evp =
-				(const struct prefix_evpn *)bgp_dest_get_prefix(
-					dest);
+	if (!rddest)
+		return 0;
```

Comment:
Seems we can even convert to void this function?

Reply:
Yes, i think so. But i had found many `delete_*()` or `update_*()` functions in bgp_evpn.c have this issue, so i stopped.
If needed, i will create new PR for it.